plugin/src/main/java/org/opensearch/ml/action/model_group/TransportUpdateModelGroupAction.java

```diff
@@ -0,0 +1,147 @@
+                        String access = (String) source.get(MLModelGroup.ACCESS);
+
+                        if (SecurityUtils.isAdmin(user) || access == "public") {
+                            updateModelGroup(modelGroupId, source, updateModelGroupInput, listener);
+                        } else if (user.getName().equals(owner.get("name"))) {
+                            if (updateModelGroupInput.getBackendRoles() != null) {
+
+                                List<String> backendRoles = user.getBackendRoles() != null ? user.getBackendRoles() : ImmutableList.of();
+
+                                if (updateModelGroupInput.getBackendRoles().size() == 0) {
+                                    updateModelGroupInput.setBackendRoles(backendRoles);
+                                }
+
+                                Boolean isRolePresent = updateModelGroupInput
+                                    .getBackendRoles()
+                                    .stream()
+                                    .allMatch(user.getBackendRoles().stream().collect(Collectors.toSet())::contains);
+
+                                if (!isRolePresent) {
+                                    log.error("Invalid Backend Roles provided in the input");
+                                    throw new IllegalArgumentException("Invalid Backend Roles provided in the input");
+                                }
+                            }
+
+                            updateModelGroup(modelGroupId, source, updateModelGroupInput, listener);
+                        } else {
+                            throw new IllegalArgumentException("User doesn't have valid privilege to perform this operation");
+                        }
+                    }
+                }, e -> {
+                    logException("Failed to Update model model", e, log);
+                    listener.onFailure(e);
+                }));
+            } catch (Exception e) {
+                logException("Failed to Update model group", e, log);
+                listener.onFailure(e);
+            }
+        }
+
+    }
+
+    private void updateModelGroup(
+        String modelGroupId,
+        Map<String, Object> source,
+        MLUpdateModelGroupInput updateModelGroupInput,
+        ActionListener<MLUpdateModelGroupResponse> listener
+    ) {
+        if (StringUtils.isNotBlank(updateModelGroupInput.getName())) {
+            source.put(MLUpdateModelGroupInput.NAME, updateModelGroupInput.getName());
+        }
+        if (StringUtils.isNotBlank(updateModelGroupInput.getDescription())) {
+            source.put(MLUpdateModelGroupInput.DESCRIPTION, updateModelGroupInput.getDescription());
+        }
+        if (updateModelGroupInput.getBackendRoles() != null) {
+            source.put(MLUpdateModelGroupInput.BACKEND_ROLES, updateModelGroupInput.getBackendRoles());
+        }
+        UpdateRequest updateModelGroupRequest = new UpdateRequest();
```

Comment:
Tags should also be able to be updated.

Reply:
Is there any relation between names of model group and its models? Also for tags, I haven't got much time to work on this. Updating a tag would require updates to its models' tags as well. Is the tags field p0?